java/azure-eventhubs/src/main/java/com/microsoft/azure/servicebus/MessageSender.java

```diff
@@ -521,8 +566,9 @@ public void onSendComplete(final Delivery delivery)
 			{
 				this.lastKnownLinkError = null;
 				this.retryPolicy.resetRetryCount(this.getClientId());
-				pendingSend.complete(null);
+
 				pendingSendWorkItem.getTimeoutTask().cancel(false);
+				pendingSend.complete(null);
```

Comment:
given, the timerTask and the actual CompletableFuture are really 2 different concepts - I would keep them as it is.